File: 1470-shuffle-the-array/1470-shuffle-the-array.kt
class Solution {
    fun shuffle(nums: IntArray, n: Int): IntArray {
        val res = IntArray(nums.size)
        for (i in 0..(n-1)) {
            res[i*2] = nums[i]
            res[i*2 + 1] = nums[i+n]
        }
        return res
    }
}

File: 1470-shuffle-the-array/1470-shuffle-the-array.js
/**
 * @param {number[]} nums
 * @param {number} n
 * @return {number[]}
 */
var shuffle = function(nums, n) {
    const res = []
    for (let i=0; i < n; i++) {
        for (let j=i; j < nums.length; j += n) {
            res.push(nums[j])
        }
    }
    return res
};

shuffle = function(nums, n) {
    const res = []
    for (let i=0; i < n; i++) {
        res.push(nums[i])
        res.push(nums[i+n])
    }
    return res
}